src/domain/core/src/repos/metadata_chain_visitor.rs

```diff
@@ -43,6 +43,9 @@ pub enum MetadataVisitorDecision {
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+// TODO: PERF: MetadataChainVisitor::visit(): use Cow<HashedMetadataBlock>.
```

Comment:
My thought was, if we see in `MetadataChainExt::accept_by_interval()` that there is only one visitor, we could pass the block and hash (`(hash, block)`) by value, instead of by reference as now (`(&hash, &block)`).

That way, if only one visitor is used, we could avoid cloning -- this only applies to those who save blocks (e.g. `SearchSingleDataBlockVisitor`).

--

If the idea seems appropriate to you after this comment, I'll update the comment to (add context) to make it clearer

Reply:
Let's not spend time on this micro-optimization until we have a clear indication that it's a performance problem